### DISCUSSION TITLE
Agent Monitor Error

### DISCUSSION THREAD

**Discussion by DarWhiteWexico** [glpi-project/glpi-agentmonitor#15] [Q&A]
When I click new ticket, i am getting "Error getting GLPI server URL! The agent must be reinstalled."

I have done multiple deletions and still get same error

**Comment by redddcyclone** [comment]
Hi @DarWhiteWexico 

Could you open regedit.exe and post a screenshot of the contents of value ```HKEY_LOCAL_MACHINE\SOFTWARE\GLPI-Agent\server```?

**Reply by DarWhiteWexico** [reply]
![image](https://github.com/user-attachments/assets/6dc3b33a-b6a7-41b8-9e1e-57e2b099c2b1)

**Reply by redddcyclone** [reply]
The error is happening because the ```server``` parameter doesn't start with ```http://``` or ```https://```

@g-bougard, is this parameter supported in this format?

**Reply by DarWhiteWexico** [reply]
Thanks, I have updated the address but it still says that reinstallation is needed 
![image](https://github.com/user-attachments/assets/b5245bd3-0ffe-40aa-b7bf-03001b1d3e86)

![image](https://github.com/user-attachments/assets/dc9bcbe1-fcea-49f4-894d-dc78fd6a5b18)

**Reply by redddcyclone** [reply]
I tried with the exact same URL in your screenshot and didn't get this error... Didn't you put any spaces on the url?

**Reply by DarWhiteWexico** [reply]
Nope, just checked and there is no white spaces anywhere in the address

**Reply by redddcyclone** [reply]
Did you close the Agent Monitor after you changed the string? The URL is obtained from the registry on Monitor startup

**Reply by DarWhiteWexico** [reply]
I am not sure I understand, when I open the agent monitor, that is where I am prompted for "New Ticket"

![image](https://github.com/user-attachments/assets/8a4fc9c7-8819-4614-b4d8-a1c156cc0d73)

Do I need to do something differnet?

**Reply by redddcyclone** [reply]
You shouldn't be prompted for the new ticket URL. This dialog shows only when you click "Settings". You need to close GLPI Agent Monitor, not only by clicking "Close" or "X" on this window, but closing it in the system tray by right clicking its icon and selecting "Exit"

**Reply by DarWhiteWexico** [reply]
So am I correct in that when I click "New Ticket" in the above image, that this does not actually redirect you to create a new ticket? What is the purpose of that button if so?

**Reply by redddcyclone** [reply]
"New Ticket" does redirect to the new ticket URL, but there is a variable that is set on the Monitor's startup if the ```server``` parameter (not the "New Ticket" parameter) is set correctly (i.e. starts with ```http://``` or ```https://```), and this variable is checked when you click the "New Ticket" button. You're getting this error because the server was not obtained correctly on the Monitor's startup. This behavior is not the best, as the new ticket URL now has a dedicated parameter, and I will fix this, but for now, as you added ```http://``` this error should not be happening anymore, but you **must** close the Monitor by doing what I said on my last reply and open it again for the error to stop showing up.

**Reply by DarWhiteWexico** [reply]
Than you for the assistance. That was exactly what the issue was. I appreciate your help. It is greatly appreciated.

I have another question, in regards to the "Force Inventory". 
Is it possible to have all computers with the agent installed to do a force inventory, so that they all update the details to glpi inventory?

Thanks again

**Reply by redddcyclone** [reply]
Well, the Monitor can't do such a thing because it does not listens to HTTP connections. I think this feature would be best suited for the ```glpi-inventory-plugin```. Actually, it's a feature that I miss on GLPI too, but while it doesn't exists, I actually fill a text file with the hostnames/IPs on each line and run ```while read in; do curl -m 2000 http://$in:62354/now; done < list.txt``` on Linux.

**Reply by g-bougard** [reply]
> The error is happening because the `server` parameter doesn't start with `http://` or `https://`
> 
> @g-bougard, is this parameter supported in this format?

Hi @redddcyclone 

yes, [glpi-agent sets "http" scheme to the url if not set](https://github.com/glpi-project/glpi-agent/blob/de5f5b515e2352c3fa864f5cbe9cd8f735ae6fff/lib/GLPI/Agent/Target/Server.pm#L50).

About the "force inventory to all agent" question. Yes, `glpi-inventory-plugin` can do the trick but this is not trivial. @redddcyclone as you seems also concerned, maybe this is the time to develop your own glpi plugin just for that :wink:, maybe with some features like a way to configure a scheduling or use dynamic groups.
